src/main/java/net/juniper/netconf/NetconfSession.java

```diff
@@ -116,6 +120,11 @@ private void sendHello(String hello) throws IOException {
         String reply = getRpcReply(hello);
         serverCapability = reply;
         lastRpcReply = reply;
+        try {
+            serverHello = Hello.from(reply);
+        } catch (final ParserConfigurationException | SAXException | XPathExpressionException e) {
+            throw new NetconfException("Invalid <hello> message from server: " + reply, e);
+        }
```

Comment:
May I ask about a pack of `final` words in classes? I can explain the usage of this word for input parameters and in global fields but why do we need to have it for every single variable? Especially in multi-catch sections because these variables are implicitly declared final. Moreover, there is an entity like effective final probably it'd be better to make java compiler works instead of developers, what do you think?

Reply:
Effectively final was introduced in Java 8. I've been developing in Java on and off since about 1.3, so it's become a fairly reflexive habit for me (*). I'm happy to remove it from where it's unnecessary if you think that's the right thing to do in this case.

(*) Kind of scary now I've written that to realise how long I've been at this! Probably spent more time developing with Java 1.6 than anything else, but that's probably a reflection of how long Java 1.6 was "current"!